luwak/src/main/java/uk/co/flax/luwak/termextractor/querytree/ConjunctionNode.java

```diff
@@ -25,62 +28,54 @@
 
 public class ConjunctionNode extends QueryTree {
 
+    private final List<QueryTree> children = new ArrayList<>();
+
     private ConjunctionNode(List<QueryTree> children) {
-        for (QueryTree child : children) {
-            this.addChild(child);
-        }
+        this.children.addAll(children);
+        this.children.sort(Comparator.comparingDouble(QueryTree::weight).reversed());
```

Comment:
This Comparator is used in two places- have you considered the Comparator making it a static final constant that can be referenced from both the constructor and advancePhase?